tools/mksurfdata_esmf/gen_mksurfdata_jobscript_single.py

```diff
@@ -43,6 +43,14 @@ def get_parser():
         dest="tasks_per_node",
         required=True,
     )
+    parser.add_argument(
+        "--machine",
+        help="""currently this recognizes cheyenne and casper (default cheyenne)""",
+        action="store",
+        dest="machine",
+        required=False,
+        default='cheyenne'
+    )
```

Comment:
There must be a way to list the valid options, rather than leaving it open ended.

Reply:
Doing this takes care of your request for a check that only the allowed machine names can be set.